hummingbot/connector/exchange/kraken/kraken_api_user_stream_data_source.py

```diff
@@ -66,30 +66,30 @@ async def get_auth_token(self) -> str:
             timeout=100
         )
 
-        async with response_coro as response:
-            if response.status != 200:
-                raise IOError(f"Error fetching Kraken user stream listen key. HTTP status is {response.status}.")
+        async with self._throttler.execute_task(CONSTANTS.GET_TOKEN_PATH_URL):
+            async with response_coro as response:
+                if response.status != 200:
+                    raise IOError(f"Error fetching Kraken user stream listen key. HTTP status is {response.status}.")
 
-            try:
-                response_json: Dict[str, Any] = await response.json()
-            except Exception:
-                raise IOError(f"Error parsing data from {url}.")
+                try:
+                    response_json: Dict[str, Any] = await response.json()
+                except Exception:
+                    raise IOError(f"Error parsing data from {url}.")
 
-            err = response_json["error"]
-            if "EAPI:Invalid nonce" in err:
-                self.logger().error(f"Invalid nonce error from {url}. " +
-                                    "Please ensure your Kraken API key nonce window is at least 10, " +
-                                    "and if needed reset your API key.")
-                raise IOError({"error": response_json})
+                err = response_json["error"]
+                if "EAPI:Invalid nonce" in err:
+                    self.logger().error(f"Invalid nonce error from {url}. " +
+                                        "Please ensure your Kraken API key nonce window is at least 10, " +
+                                        "and if needed reset your API key.")
+                    raise IOError({"error": response_json})
 
-            return response_json["result"]["token"]
+                return response_json["result"]["token"]
 
-    async def listen_for_user_stream(self, ev_loop: asyncio.BaseEventLoop, output: asyncio.Queue):
+    async def listen_for_user_stream(self, ev_loop: asyncio.AbstractEventLoop, output: asyncio.Queue):
         while True:
             try:
-                async with websockets.connect(KRAKEN_WS_URL) as ws:
-                    ws: websockets.WebSocketClientProtocol = ws
-
+                async with self._throttler.execute_task(CONSTANTS.WS_CONNECTION_LIMIT_ID):
+                    ws = await websockets.connect(CONSTANTS.WS_AUTH_URL)
```

Comment:
If we are not using `async with websockets.connect ...` then we need to add a `finally`component to the try to close the ws connection if it was stablished.